finagle/buoyant/src/main/scala/com/twitter/finagle/buoyant/TlsClientConfig.scala

```diff
@@ -31,33 +31,17 @@ case class TlsClientConfig(
       Stack.Params.empty + Transport.ClientSsl(Some(tlsConfig)) +
         SslClientEngineFactory.Param(Netty4ClientEngineFactory())
 
-    case TlsClientConfig(_, _, Some(cn), certs, clientAuth, enabledProtocols) =>
-      // map over the optional certs parameter - we want to pass
-      // `TrustCredentials.CertCollection` if we were given a list of certs,
-      // but `TrustCredentials.Unspecified` (rather than an empty cert
-      // collection file) if we were not.
-      val credentials = certs.map { certs =>
-        // a temporary file to hold the collection of certificates
-        val certCollection = File.createTempFile("certCollection", null)
-        // open the cert paths as Streams...
-        val f = new FileOutputStream(certCollection)
-        for {
-          cert <- certs
-          certStream = new FileInputStream(cert)
-        } { // ...and copy the certs into the cert collection
-          // TODO: can this be made more concise with scala.io?
-          StreamIO.copy(certStream, f)
+    case TlsClientConfig(_, _, Some(cn), certs, bundle, clientAuth, enabledProtocols) =>
```

Comment:
I'd prefer to be a bit more defensive here and throw an IllegalArgumentException if trustCerts and trustCertsBundle are both set.  Then we can handle them in separate cases:

```
case TlsClientConfig(_, _, Some(cn), Some(certs), Some(bundle), _, _) =>
  throw ...
case TlsClientConfig(_, _, Some(cn), Some(certs), None, _, _) =>
  // use certs
case TlsClientConfig(_, _, Some(cn), None, Some(bundle), _, _) =>
  // use bundle
case TlsClientConfig(_, _, Some(cn), None, None, _, _) =>
  // use Unspecified
```